### PR TITLE
Implemented basic traffic injection for BaseBackend

### PR DESCRIPTION
**Pull request by Fusaaaann**
# Description

This PR introduces a Python-native implementation for multi-threaded traffic injection targeting backends in `enova.serving`. A POST endpoint, `/v1/inject`, is set up at server startup. Currently, traffic injection is supported only for servers compatible with the OpenAI API.

### Example Request

To use the injection interface, send a request like the following:

```json
{
    "distribution": "poisson",
    "tps_mean": 10,
    "duration": 10,
    "duration_unit": "s",
    "data_set": "arc"
}
```

Value of key "data_set" can be replaced with any of [other supported datasets](https://github.com/Emerging-AI/ENOVA/tree/9348e3143ff92a8e8833076534b995fb7e6c00b9/enova/template/deployment/docker-compose/traffic-injector/data). The variables of each row will be converted to the Completions API request model at runtime of VanillaTrafficInjector.

### Supported Distributions

The traffic injection supports both Gaussian and Poisson distributions for request generation:

- **Gaussian Distribution**: Specify the mean and standard deviation of transactions per second (TPS).

```jsonc
{
    // ...
    "distribution": "gaussian",
    "tps_mean": 10,
    "tps_std": 1,
    // ...
}
```

- **Poisson Distribution**: Specify only the mean TPS.

```jsonc
{
    // ...
    "distribution": "poisson",
    "tps_mean": 10,
    // ...
}
```

This implementation allows for flexible testing scenarios by simulating different traffic patterns on the server.

### CODE REVIEW

File: enova/serving/backend/injector.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            raise TestStartError(f"VanillaTrafficInjector: argument 'timer' provided is invalid, got {timer} instead")
+        def make_request(request_func, url, headers, json_data):
+            request_func(url, headers=headers, json=json_data)
+        threads_started = []
+        while True:
+            try:
+                sleep_duration = next(delay_iter)
+            except StopIteration:
+                break
+            thread = threading.Thread(target=make_request, args=(request_method, f"http://{host}:{port}{path}", headers, next(dataset_iter)))
```
Comment (kyokagong) [inline enova/serving/backend/injector.py]: 这里要改成 threadpool哈。 防止太密集打爆服务线程。

File: enova/serving/backend/injector.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+        # initialize timer
+        if timer['type'] == Distribution.POISSON.value:
+            delay_iter = iter(np.random.poisson(timer["lambda"],duration))
+        elif timer['type'] == Distribution.NORMAL.value:
+            delay_iter = iter(np.random.normal(timer["mean"], timer["std"], duration))
+        else:
+            raise TestStartError(f"VanillaTrafficInjector: argument 'timer' provided is invalid, got {timer} instead")
+        def make_request(request_func, url, headers, json_data):
+            request_func(url, headers=headers, json=json_data)
```
Comment (kyokagong) [inline enova/serving/backend/injector.py:96]: 增加 try exception 捕捉错误并打印哈。 不然出错都不知道为什么啦

Comment (Fusaaaann) [reply enova/serving/backend/injector.py:96]: 这方面我计划是在调用injector这层捕捉： https://github.com/Emerging-AI/ENOVA/blob/9b6dcac6157a099494834952033aa6931bcc9fc2/enova/serving/backend/base.py#L87C1-L103C29

File: enova/serving/backend/injector.py (hunk 1 above)
Comment (Fusaaaann) [reply enova/serving/backend/injector.py]: 已完成： [`5ff101a` (#43)](https://github.com/Emerging-AI/ENOVA/pull/43/commits/5ff101a817a8d6383ffccfe7700769fde8d7fd5c)

**Pull request merged**